pkg/config/config.go

```diff
@@ -449,17 +449,15 @@ func validateModuleUseReferences(p modulePath, mod Module, bp Blueprint) error {
 }
 
 func checkBackend(b TerraformBackend) error {
-	const errMsg = "can not use variables in terraform_backend block, got '%s=%s'"
-	// TerraformBackend.Type is typed as string, "simple" variables and HCL literals stay "as is".
-	if hasVariable(b.Type) {
-		return fmt.Errorf(errMsg, "type", b.Type)
-	}
-	if _, is := IsYamlExpressionLiteral(cty.StringVal(b.Type)); is {
-		return fmt.Errorf(errMsg, "type", b.Type)
+	err := errors.New("can not use expressions in terraform_backend block")
+	val, perr := parseYamlString(b.Type)
```

Comment:
Shoudn't we check here if perr is nil or not before calling IsExpressionValue ?

Reply:
`IsExpressionValue` can safely take `cty.NilVal`, so there is no reason for separate check.